protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,494 @@
+        _burnAndNotify(amount, _config.recipient, useContractEth);
+    }
+
+    /**
+     * @notice Get spice auction config for an auction
+     * @param auctionId Id of auction
+     */
+    function getAuctionConfig(uint256 auctionId) external view override returns (SpiceAuctionConfig memory) {
+        return auctionConfigs[auctionId];
+    }
+
+    /**
+     * @notice Get auction token for current epoch
+     * @return Auction token
+     */
+    function getAuctionTokenForCurrentEpoch() external override view returns (address) {
+        SpiceAuctionConfig memory config = auctionConfigs[_currentEpochId];
+        return config.isTempleGoldAuctionToken ? templeGold : spiceToken;
+    }
+
+    /**
+     * @notice Get current epoch
+     * @return Epoch Id
+     */
+    function currentEpoch() external view override returns (uint256) {
+        return _currentEpochId;
+    }
+
+    /**
+     * @notice Get claimable amount for an epoch
+     * @dev function will return claimable for epoch. This can change with more user deposits
+     * @param depositor Address to check amount for
+     * @param epochId Epoch id
+     * @return Claimable amount
+     */
+    function getClaimableForEpoch(address depositor, uint256 epochId) external override view returns (uint256) {
+        if (claimed[depositor][epochId]) { return 0; }
+        uint256 bidTokenAmount = depositors[depositor][epochId];
+        if (bidTokenAmount == 0 || epochId > _currentEpochId) { return 0; }
+        EpochInfo memory info = epochs[epochId];
+        return bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+    }
+
+    function _getBidAndAuctionTokens(
+        SpiceAuctionConfig storage _config
+    ) private view returns (address bidToken, address auctionToken) {
+        (bidToken, auctionToken) = _config.isTempleGoldAuctionToken
+            ? (spiceToken, templeGold)
+            : (templeGold, spiceToken);
+    }
+
+    function _burnAndNotify(uint256 amount, address from, bool useContractEth) private {
+        // pull funds from bids recipient (set in config)
+        IERC20(templeGold).safeTransferFrom(from, address(this), amount);
+        // burn directly and call TempleGold to update circulating supply
+        if (block.chainid == _mintChainId) {
+            ITempleGold(templeGold).burn(amount);
+            return;
+        }
+        bytes memory options = OptionsBuilder.newOptions().addExecutorLzReceiveOption(lzReceiveExecutorGas, 0);
+        SendParam memory sendParam = SendParam(
+            _arbitrumOneLzEid, //<ARB_EID>,
+            bytes32(uint256(uint160(address(0)))), // bytes32(address(0)) to burn
+            amount,
+            0,
+            options,
+            bytes(""), // compose message
+            ""
+        );
+        MessagingFee memory fee = ITempleGold(templeGold).quoteSend(sendParam, false);
+        if (useContractEth && address(this).balance < fee.nativeFee) {
```

Comment:
just in case, note that `address(this).balance` would also include the value of `msg.value` when reaching this piece of code. 

I think the rest of the logic is not affected

Reply:
If you really want to enforce it, you can `require(msg.value == 0)`, but I don't think it makes sense. Allowing sending `msg.value` while passing `useContractEth=true` is fine, so that the caller can send extra ether if the contract needs it